.github/workflows/test_code.yml

```diff
@@ -21,7 +21,7 @@ jobs:
     - name: clang-format
       run: cmake --build build --target format
     - name: check formatting
-      run: git diff --quiet
+      run: git diff --quiet || ( git diff && false )
```

Comment:
Not sure if we want this. The output of the command isn't relevant at all. `make format` takes care of fixing it.
But if we do, one call to `git diff --exit-code` would be enough

Reply:
It's always better to see what are the exact "complaints" of the formatter. In my case it was just a single line to fix, so it was even easier to edit by hand than running `make format` locally.

Reply:
OK, so let's use `git diff --exit-code` then

Reply:
How about that?

```
    - name: check formatting
      run: git diff --exit-code || ( >&2 echo "Please run 'make format' to fix these issues automatically." && false )
```

Reply:
Sure :>